### DISCUSSION TITLE
Why not the C# 8 Indices and ranges does not implements like Python slice?

### DISCUSSION THREAD

**Discussion by GF-Huang** [dotnet/csharplang#4157] [Q&A]
#### Python no exception:
![image](https://user-images.githubusercontent.com/4510984/99734831-75b83b80-2afe-11eb-9fc0-97f6a873a7fd.png)

#### C# give me out of range:
![image](https://user-images.githubusercontent.com/4510984/99734966-b912aa00-2afe-11eb-8528-bfa9e77522f7.png)

I know that just a syntactic sugar of `string.Substring`, but why not implements like python? Even if need to inject length checking.

**Comment by Unknown6656** [comment]
There are several arguments for your proposal, as well as for the current implementation.
However, changing it now would be a **breaking change**, so this proposal would never be accepted.

Further, the correct place for this proposal is https://github.com/dotnet/runtime. This repo is only dedicated to the C# syntax, not the runtime's behaviour.

**Comment by CyrusNajmabadi** [comment]
> but why not implements like python? 

Because it seemed worse than the semantics we chose
